elixir/wrapper/module.py

```diff
@@ -0,0 +1,213 @@
+from collections import defaultdict
+from copy import copy
+from functools import partial
+from typing import Iterable
+
+import torch
+import torch.nn as nn
+from torch.distributed import ProcessGroup
+from torch.utils._pytree import tree_map
+
+from elixir import gpu_dev
+from elixir.chunk import Chunk, ChunkFetcher, ChunkGroup, MemoryPool, TensorState
+from elixir.chunk.scheduler import FIFOScheduler, PrefetchScheduler
+from elixir.hook import HookParam
+from elixir.parameter import FakeTensor, OutplaceTensor
+from elixir.search import SearchResult
+
+
+def get_param_optim_data(param_data: torch.Tensor, param_dtype: torch.dtype):
+    param_data = param_data.to(gpu_dev())
+    optim_data = param_data.clone() if param_data.dtype == torch.float else param_data.float()
+    param_data = param_data.to(param_dtype)
+    return param_data, optim_data
+
+
+class ElixirModule(nn.Module):
+
```

Comment:
Do take of documentation for maintainability. Add docstrings whenever appropriate.